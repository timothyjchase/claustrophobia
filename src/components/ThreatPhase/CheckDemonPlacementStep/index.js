import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Button, Checkbox } from 'semantic-ui-react'

class CheckDemonPlacementStep extends Component {
  constructor(props) {
    super(props)
    this.state = { legalPlacement: false }
  }

  render() {
    const { completeThreatDemonPlacementStep } = this.props
    const { legalPlacement } = this.state

    return (
      <div>
        <Checkbox
          label="At least 1 legal placement for a Demon within 4 tiles of a Human warrior?"
          onChange={() => this.setState({ legalPlacement: !legalPlacement })}
          checked={legalPlacement}
        />
        <br />
        <br />
        <Button.Group vertical fluid>
          <Button
            primary
            icon="play"
            content="Next"
            onClick={() => completeThreatDemonPlacementStep(legalPlacement)}
          />
        </Button.Group>
      </div>
    )
  }
}

CheckDemonPlacementStep.propTypes = {
  completeThreatDemonPlacementStep: PropTypes.func.isRequired,
}

export default CheckDemonPlacementStep
