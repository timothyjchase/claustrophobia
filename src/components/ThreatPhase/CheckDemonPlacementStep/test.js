import React from 'react'
import CheckDemonPlacementStep from './index'

describe('<CheckDemonPlacementStep />', () => {
  describe('with basic props', () => {
    const props = {
      completeThreatDemonPlacementStep: jest.fn(),
    }
    const component = shallow(<CheckDemonPlacementStep {...props} />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
