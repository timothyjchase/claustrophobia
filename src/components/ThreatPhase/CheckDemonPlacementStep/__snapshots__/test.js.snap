// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckDemonPlacementStep /> with basic props should match snapshot 1`] = `
<div>
  <Checkbox
    checked={false}
    label="At least 1 legal placement for a Demon within 4 tiles of a Human warrior?"
    onChange={[Function]}
    type="checkbox"
  />
  <br />
  <br />
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Next"
      icon="play"
      onClick={[Function]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;
